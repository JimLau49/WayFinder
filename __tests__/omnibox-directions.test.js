import React from "react";
import OmniboxDirections from "../src/components/search/omnibox-directions.component";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";

let mockPOIs = [
  {
    id: "c4541028-739d-4651-b9f5-dc4d54054807",
    displayName: "Stairs 2",
    description: "",
    location: {
      latitude: 45.497482,
      longitude: -73.579034
    },
    buildingId: "BuildingId.H",
    level: 8,
    category: "Stairs"
  },
  {
    id: "f5a7d95c-170b-48c9-aac3-e9faafa6033d",
    displayName: "Stairs 3",
    description: "",
    location: {
      latitude: 45.497253,
      longitude: -73.579251
    },
    buildingId: "BuildingId.H",
    level: 8,
    category: "Stairs"
  }
];

describe("OmniboxDirections component", () => {
  it("should match snapshot", () => {
    const tree = renderer
      .create(
        <OmniboxDirections
          initialLocation={mockPOIs[0]}
          destination={mockPOIs[1]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should run provided function on press", () => {
    const mockSetDestination = jest.fn();

    const wrapper = shallow(
      <OmniboxDirections
        setDestination={mockSetDestination}
        initialLocation={mockPOIs[0]}
        destination={mockPOIs[1]}
      />
    );

    const backArrow = wrapper.find({ testID: "backArrow" });
    backArrow.simulate("press");
    expect(mockSetDestination).toHaveBeenCalledTimes(1);
  });

  it("should call a function on user input", () => {
    const mockQueryText = jest.fn();

    const wrapper = shallow(
      <OmniboxDirections
        initialLocation={mockPOIs[0]}
        destination={mockPOIs[1]}
        queryText={mockQueryText}
      />
    );

    const textInput = wrapper.find({ testID: "searchInputInitialLocation" });
    textInput.simulate("changeText", "H805");
    expect(mockQueryText).toHaveBeenCalledTimes(1);
    expect(mockQueryText.mock.calls[0][0]).toBe("H805");
  });

  it("should call a function on user input", () => {
    const mockQueryText = jest.fn();

    const wrapper = shallow(
      <OmniboxDirections
        initialLocation={mockPOIs[0]}
        destination={mockPOIs[1]}
        queryText={mockQueryText}
      />
    );

    const textInput = wrapper.find({
      testID: "searchInputDestinationLocation"
    });
    textInput.simulate("changeText", "H805");
    expect(mockQueryText).toHaveBeenCalledTimes(1);
    expect(mockQueryText.mock.calls[0][0]).toBe("H805");
  });
});
