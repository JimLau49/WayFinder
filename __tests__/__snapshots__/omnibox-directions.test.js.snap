// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OmniboxDirections component should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
        "height": 235,
        "position": "absolute",
        "top": 0,
        "width": 750,
        "zIndex": 1,
      },
      undefined,
      Object {
        "height": 260,
      },
    ]
  }
>
  <View
    style={
      Object {
        "marginLeft": 15,
        "marginRight": 15,
        "paddingBottom": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="backArrow"
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 36,
        }
      }
    >
      <Image
        source={1}
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          selectTextOnFocus={true}
          style={
            Object {
              "borderColor": "#DEDEDE",
              "borderWidth": 1,
              "fontSize": 16,
              "height": 38,
              "marginBottom": 15,
              "marginLeft": 10,
              "paddingLeft": 10,
            }
          }
          testID="searchInputInitialLocation"
          underlineColorAndroid="transparent"
          value="Stairs 2"
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          selectTextOnFocus={true}
          style={
            Object {
              "borderColor": "#DEDEDE",
              "borderWidth": 1,
              "fontSize": 16,
              "height": 38,
              "marginBottom": 15,
              "marginLeft": 10,
              "paddingLeft": 10,
            }
          }
          testID="searchInputDestinationLocation"
          underlineColorAndroid="transparent"
          value="Stairs 3"
        />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#DEDEDE",
          "borderWidth": 1,
          "height": 35,
          "marginBottom": 20,
          "marginLeft": 38,
        }
      }
    >
      <Text
        adjustFontSizeToFit={true}
        allowFontScaling={false}
        numberOfLines={1}
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 13,
          }
        }
      >
        Departure time 9AM
      </Text>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onOrientationChange={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
            "landscape-left",
            "landscape-right",
          ]
        }
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F5FCFF",
                  "justifyContent": "center",
                  "padding": 0,
                },
                Object {
                  "width": 750,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#CCCFD6",
                    "borderBottomWidth": 0,
                    "borderColor": "#CCCFD6",
                    "borderTopColor": "lightgrey",
                    "borderTopWidth": 0.5,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "padding": 0,
                  },
                  undefined,
                  Object {
                    "width": 750,
                  },
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#006BFF",
                        "padding": 8,
                      },
                      undefined,
                      Object {
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  Cancel
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#006BFF",
                        "padding": 8,
                      },
                      undefined,
                      Object {
                        "textAlign": "right",
                      },
                    ]
                  }
                >
                  Done
                </Text>
              </View>
            </View>
            <View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#FFFFFF",
                    },
                    undefined,
                    Object {
                      "width": 750,
                    },
                  ]
                }
              >
                <RCTPicker
                  items={
                    Array [
                      Object {
                        "label": "Departure time 9AM",
                        "textColor": undefined,
                        "value": 0,
                      },
                      Object {
                        "label": "bithc",
                        "textColor": undefined,
                        "value": 1,
                      },
                      Object {
                        "label": "please",
                        "textColor": undefined,
                        "value": 2,
                      },
                      Object {
                        "label": "dlsfjhlds",
                        "textColor": undefined,
                        "value": 3,
                      },
                      Object {
                        "label": "bisaljdfhlathc",
                        "textColor": undefined,
                        "value": 4,
                      },
                      Object {
                        "label": "bijsdfa lsdkfthc",
                        "textColor": undefined,
                        "value": 5,
                      },
                      Object {
                        "label": "biasjdk fdsthc",
                        "textColor": undefined,
                        "value": 6,
                      },
                    ]
                  }
                  onChange={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  selectedIndex={0}
                  style={
                    Array [
                      Object {
                        "height": 216,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text />
      <Text />
      <Text />
      <Text />
      <Image
        source={1}
        style={
          Object {
            "marginLeft": 15,
          }
        }
      />
    </View>
  </View>
</View>
`;
